docs/DeveloperGuide.md

```diff
@@ -6,8 +6,70 @@
 
 ## Design & implementation
```

Comment:
The developer guide does not seem to be building to the project website https://ay2324s2-cs2113-w12-3.github.io/tp/DeveloperGuide.html